File: nodejs/database/tokens.ts
import { DbData, db } from './db.ts';

const addTokenStatement = db.prepare(`INSERT INTO tokens(data) VALUES (?);`);
const getTokenStatement = db.prepare(`SELECT data FROM tokens;`);

export function addToken(token: string): void {
	addTokenStatement.run(token);
}

export function getToken(): string|null {
	const data = getTokenStatement.get() as DbData|undefined;
	if (data === undefined) return null;
	return data.data;
}

File: golang/pkg/database/tokens.go
package database

func (db *DB) AddToken(token string) error {
	_, err := db.Exec(`INSERT INTO tokens(data) VALUES (?);`, token)
	return err
}

func (db *DB) GetToken() (string, error) {
	var token string
	err := db.QueryRow(`SELECT data FROM tokens;`).Scan(&token)
	return token, err
}
